internal/chain/reward.go

```diff
@@ -81,40 +83,35 @@ func lookupDelegatorShares(
 ) (map[common.Address]numeric.Dec, error) {
 	epoch := snapshot.Epoch
 	validatorSnapshot := snapshot.Validator
-	key := fmt.Sprintf("%s-%s", epoch.String(), validatorSnapshot.Address.Hex())
 
-	shares, err, _ := delegateShareCache.Do(
-		key, func() (interface{}, error) {
-			result := map[common.Address]numeric.Dec{}
+	// Look up
+	key := fmt.Sprintf("%s-%s", epoch.String(), validatorSnapshot.Address.Hex())
 
-			totalDelegationDec := numeric.NewDecFromBigInt(validatorSnapshot.TotalDelegation())
-			if totalDelegationDec.IsZero() {
-				utils.Logger().Info().
-					RawJSON("validator-snapshot", []byte(validatorSnapshot.String())).
-					Msg("zero total delegation during AddReward delegation payout")
-				return result, nil
-			}
+	if b, ok := delegateShareCache.Get(key); ok {
+		return b.(map[common.Address]numeric.Dec), nil
+	}
 
-			for i := range validatorSnapshot.Delegations {
-				delegation := validatorSnapshot.Delegations[i]
-				// NOTE percentage = <this_delegator_amount>/<total_delegation>
-				percentage := numeric.NewDecFromBigInt(delegation.Amount).Quo(totalDelegationDec)
-				result[delegation.DelegatorAddress] = percentage
-			}
+	// If not found, construct
+	result := map[common.Address]numeric.Dec{}
```

Comment:
will the map cause concurrency issue?

Reply:
do you need to protect it?

Reply:
this map is atomically created and only read, never updated. So no concurrency issue.